components/SearchResult.tsx

```diff
@@ -1,24 +1,29 @@
-import { useQuery } from "@tanstack/react-query";
+import { useInfiniteQuery } from "@tanstack/react-query";
 import Result from "./Result";
 import { Album, SearchResponseSchema } from "@/schemas/collections.schemas";
 import ResultLoading from "./ResultLoading";
 
 export default function SearchResult({ searchValue }: { searchValue: string }) {
-  const { data, isError, isLoading } = useQuery({
+  const { data, error, isFetching, fetchNextPage } = useInfiniteQuery({
     queryKey: ["search", searchValue],
-    queryFn: async () => {
+    queryFn: async ({ pageParam }) => {
       const params = new URLSearchParams({
-        search: searchValue,
+        search: searchValue ? searchValue : "flume",
+        page: String(pageParam),
       });
+
       const res = await fetch("/api/search?" + params);
       const data = await res.json();
       const parsedData = SearchResponseSchema.parse(data);
-
       return parsedData;
     },
+    initialPageParam: 1,
+    getNextPageParam: (lastPage, allPages, lastPageParam) => {
```

Comment:
hmm not sure where you're seeing this has a 3rd argument, [here's the docs for it](https://tanstack.com/query/v4/docs/react/reference/useInfiniteQuery)

![CleanShot 2023-11-04 at 15 57 57](https://github.com/Rmaki27/collections/assets/37197876/8704e5ef-8237-4a61-92be-0a8e4ddefa42)

Discogs [has a page field in their return response](https://www.discogs.com/developers#page:home,header:home-pagination), getNextPageParam should look at that to know what page is next